osidb/tests/test_cvss.py

```diff
@@ -173,13 +173,51 @@ def test_validate_rh_cvss3_and_impact(self, impact, vector, should_alert):
 
         cvss.save()
         alerts = [
-            "zero_CVSSv3_score_with_set_impact",
-            "nonzero_CVSSv3_score_with_unset_impact",
+            "set_impact_with_zero_CVSSv3_score",
+            "unset_impact_with_nonzero_CVSSv3_score",
         ]
 
         if should_alert:
-            assert cvss.valid_alerts.filter(name=should_alert).exists()
+            assert flaw.valid_alerts.filter(name=should_alert).exists()
         else:
             assert not any(
-                [cvss.valid_alerts.filter(name=alert).exists() for alert in alerts]
+                [flaw.valid_alerts.filter(name=alert).exists() for alert in alerts]
             )
+
+    @pytest.mark.enable_signals
+    @pytest.mark.parametrize(
+        "impact,vector,should_alert",
+        [
+            # score 0.0
+            (Impact.LOW, "CVSS:3.1/AV:N/AC:H/PR:N/UI:N/S:U/C:N/I:N/A:N", True),
+            # score 0.0
+            (Impact.NOVALUE, "CVSS:3.1/AV:N/AC:H/PR:N/UI:N/S:U/C:N/I:N/A:N", False),
+        ],
+    )
+    def test_validate_rh_cvss3_and_impact_deadlock(self, impact, vector, should_alert):
+        """
+        Test that the RH CVSSv3 score can be changed from zero score and unset flaw impact.
+        This is here to make sure that a deadlock does not occur when trying to escape from
+        an invalid RH CVSSv3 score / impact state.
+        """
+
+        flaw = FlawFactory(impact=impact)
+        cvss = FlawCVSSFactory.build(
+            flaw=flaw,
+            issuer=FlawCVSS.CVSSIssuer.REDHAT,
+            version=FlawCVSS.CVSSVersion.VERSION3,
+            vector=vector,
+        )
+
+        assert cvss.save() is None
+        cvss.vector = "CVSS:3.1/AV:P/AC:L/PR:L/UI:R/S:C/C:H/I:H/A:H"  # score 7.2
+        cvss.save()
+
+        alerted = flaw.valid_alerts.filter(
+            name="unset_impact_with_nonzero_CVSSv3_score"
+        ).exists()
+
+        # Alert status should flip after changing the impact
+        should_alert = not should_alert
```

Comment:
The flip seems to me to decrease the test readability. The value is only used once so I do not see a reason to flip it. I would recommend to just define it in other way when the flip is not necessary and it holds the information about what we expect right from the beginning.